### DISCUSSION TITLE
SK.Quit()

### DISCUSSION THREAD

**Discussion by paulmelis** [StereoKit/StereoKit#327] [Q&A]
Just to verify, I've got a button hooked up to calling `SK.Quit()`. And although the application indeed exits the window representation (for lack of a better word) does stay around and I need to manually close it with the X button. Is that as expected?

**Comment by maluoi** [comment]
I might need a bit more context here, SK.Quit should be pretty solid! What it _does_ do is cause SK.Step to return false, which will then exit the main loop. If there's something else keeping the application alive, or you've got a different main loop, this may not have the effect you're looking for though.

**Reply by paulmelis** [reply]
Stripped to the relevant parts I'm using this:

```
    class Application
    {
        static void Main(string[] args)
        {
            SKSettings settings = new SKSettings
            {
                appName = "HoloLensDemo2",
                assetsFolder = "Assets",
            };

            if (!SK.Initialize(settings))
                Environment.Exit(1);

            Application app = new Application();
            app.Run();

            SK.Shutdown();
        }

        void Run()
        {
            while (SK.Step(() =>
            {
                  UI.WindowBegin("Window", ref windowPose, new Vec2(20, 0) * U.cm, UIWin.Normal);
                  if (UI.Button("Q"))
                       SK.Quit();
                  UI.WindowEnd()
            }));
        }
    }
```

I do have a thread running for fetching network updates, but its `IsBackground` is set to `true`, so I would expect it not to block the exit. Just verified that by not starting that thread, and still get the window remaining after pressing quit.

Also checked that the same app in x64 mode in the simulator fully exits and it does. It's just that on HoloLens the window remains, strangely enough.

Another weird thing: after quitting the app the first time and, instead of closing the window, I activate it again the app restarts! So it seems the leftover window is not really the dormant form as it usually is, but I don't really understand what is going on.

**Reply by maluoi** [reply]
Oh! I think I know what you're referring to. UWP is one of those platforms that doesn't quite believe in closing apps, it just sends them to background and occasionally can just rehydrate them, or shut them down completely if it's low on RAM. 

The "window" in the shell is actually more like an app icon in this case. For a complete application, you would likely replace this window with a [3D app launcher](https://docs.microsoft.com/en-us/windows/mixed-reality/distribute/implementing-3d-app-launchers) instead!
